### ISSUE TITLE
Movement Code

### ISSUE THREAD

**Issue by SalmonSmokedSalmon** [SalmonSmokedSalmon/Petunia-v.01#1]
This looks good Matt, the only thing is I think we want to separate the input for the movement and the movement itself. I'm going to plug away at it tonight and upload an event bus movement system with hopefully a form of dialogue tree as well. 

Edit#1: Let's see what other mechanics Ryan wants to implement and start going with those as well.

**Comment by coachmb** [comment]
No worries. Plugging away at the inventory system. I'm probably just rusty with this, but its more complicated than I first thought

**Comment by SalmonSmokedSalmon** [comment]
Yeah, it's not the same as C# either which is where I'm running into issues. 

Are you building out a UI first or trying to script items?
